installscripts/Installer.sh

```diff
@@ -61,99 +41,230 @@ spin_wheel()
+  spin_wheel $! "Installing git"
+
+  # Download and Install java
+  curl -v -j -k -L -H "Cookie: oraclelicense=accept-securebackup-cookie" http://download.oracle.com/otn-pub/java/jdk/8u131-b11/d54c1d3a095b4ff2b6607d096fa80163/jdk-8u131-linux-x64.rpm -o jdk-8u112-linux-x64.rpm >>$LOG_FILE 2>&1&
+  spin_wheel $! "Downloading java"
+
+  sudo rpm -ivh --force ./jdk-8u112-linux-x64.rpm >>$LOG_FILE 2>&1&
+  spin_wheel $! "Installing java"
+
+  sudo rm -rf jdk-8u112-linux-x64.rpm
+
+  # Download and Install unzip
+  sudo yum install -y unzip >>$LOG_FILE 2>&1&
+  spin_wheel $! "Installing unzip"
+
+  # Create a temporary folder .
+  # Here we will have all the temporary files needed and delete it at the end
+  sudo rm -rf $INSTALL_DIR/jazz_tmp
+  mkdir $INSTALL_DIR/jazz_tmp
+
+  # Download and Install awscli
+  sudo curl -L https://s3.amazonaws.com/aws-cli/awscli-bundle.zip -o $INSTALL_DIR/jazz_tmp/awscli-bundle.zip >> $LOG_FILE 2>&1 &
+  spin_wheel $! "Downloading  awscli bundle"
+  sudo rm -rf $INSTALL_DIR/jazz_tmp/awscli-bundle
+  sudo unzip $INSTALL_DIR/jazz_tmp/awscli-bundle.zip -d $INSTALL_DIR/jazz_tmp>>$LOG_FILE 2>&1 &
+  spin_wheel $! "Unzipping  awscli bundle"
+  sudo rm -rf /usr/local/aws
+  sudo rm -f /usr/local/bin/aws
+
+  cd $INSTALL_DIR/jazz_tmp/awscli-bundle/
+  sudo ./install -i /usr/local/aws -b /usr/local/bin/aws >>$LOG_FILE 2>&1 &
+  spin_wheel $! "Installing  awscli bundle"
+  cd $INSTALL_DIR/
+
+  #Download and Install Terraform
+  sudo curl -v -L https://releases.hashicorp.com/terraform/0.9.11/terraform_0.9.11_linux_amd64.zip?_ga=2.191030627.850923432.1499789921-755991382.1496973261 -o $INSTALL_DIR/jazz_tmp/terraform.zip >>$LOG_FILE 2>&1 &
+  spin_wheel $! "Downloading terraform"
+  sudo unzip -o $INSTALL_DIR/jazz_tmp/terraform.zip -d /usr/bin>>$LOG_FILE 2>&1 &
+  spin_wheel $! "Installing terraform"
+
+  #Downloading and Install atlassian-cli
+  sudo curl -L https://bobswift.atlassian.net/wiki/download/attachments/16285777/atlassian-cli-6.7.1-distribution.zip -o $INSTALL_DIR/jazz_tmp/atlassian-cli-6.7.1-distribution.zip >>$LOG_FILE 2>&1 &
+  spin_wheel $! "Downloading atlassian-cli"
+  sudo unzip -o $INSTALL_DIR/jazz_tmp/atlassian-cli-6.7.1-distribution.zip  >>$LOG_FILE 2>&1 &
+  spin_wheel $! "Installing atlassian-cli"
+
+  #Get Jazz Installer code base
+  sudo rm -rf jazz-installer
+  git clone -b $JAZZ_BRANCH https://github.com/tmobile/jazz-installer.git >>$LOG_FILE 2>&1 &
+  spin_wheel $! "Downloading jazz Installer"
+
+  #Download and Install wget
+  sudo yum install -y wget >>$LOG_FILE 2>&1 &
+  spin_wheel $! "Downloading and install wget"
+
+  #Download and install pip
+  sudo wget -q https://bootstrap.pypa.io/get-pip.py
+  sudo python get-pip.py >>$LOG_FILE 2>&1 &
+  spin_wheel $! "Downloading and install pip"
+
+  #Download and install paramiko
+  sudo pip install paramiko >>$LOG_FILE 2>&1 &
+  spin_wheel $! "Downloading and install paramiko"
+}
+
+function install_packages_verbose () {
```

Comment:
Rather than creating an entirely separate `verbose` function, you should be able to do the package install in 1 function, using `| tee -a $LOGFILE` like you are doing here, but selectively hiding the console echo for `tee` stuff _**if**_ the verbose variable is set.

See here for example: https://unix.stackexchange.com/a/28761

So in the following script, if `verbose` is 0, then `tee` creates the log file, but does not echo to the CLI. If `verbose` is 1, then tee creates the log file and also echoes to the CLI. So you can remove `install_packages_verbose` and just use `tee` everywhere, and toggle the console printing based on your variable.

```
#!/bin/bash

verbose=1

exec 3>&1
exec 4>&2

if ((verbose)); then
  echo "verbose=1"
else
  echo "verbose=0"
  exec 1>/dev/null
  exec 2>/dev/null
fi


du | tee -a log.txt
```

Reply:
Conditionally swapping the default linux file descriptors stdout(1) and stderr(2) with /dev/null OR with newer file descriptors(3 and 4) is a great idea. 

However, this approach was intentionally dropped in our use-case. Let me try an elaborate:

1. Let's assume the verbosity selected by user is "0". That would take the control to else condition in the above sample code snippet. This in turn redirects the stdout and stderr of the "entire shell" to /dev/null. Any command being executed after this step will all redirect their outputs to /dev/null and the user will not see ANY console outputs hereafter. 
2. We do have a tiny spin wheel which rotates even the user selects non-verbose mode(0). This spin wheel won't appear on console and the user will get to see nothing on the console(Truly non-verbose) :simple_smile:

Please let me know your thoughts.